src/prism.c

```diff
@@ -6817,69 +6821,68 @@ lex_identifier(pm_parser_t *parser, bool previous_command_start) {
         }
     }
 
-    if (parser->lex_state != PM_LEX_STATE_DOT) {
+    if ((parser->lex_state != PM_LEX_STATE_DOT) && (current_start + width <= end)) {
```

Comment:
Shouldn't this always be the case? If we already got this far into the identifier, then `current_start + width` would necessarily have been before the end.

Reply:
Let me try.

Reply:
OK, I think you are right, I removed that test in https://github.com/ruby/prism/pull/1849/commits/23a68dcda2493b95f9be43de9250d4f9e04e39e0